sympy/integrals/intpoly.py

```diff
@@ -0,0 +1,448 @@
+                            continue
+                    for univariate in monomial.args:
+                        term_type = len(univariate.args)
+                        if term_type == 0 and univariate.is_Symbol:
+                            degree += 1
+                        elif term_type == 2:
+                            degree += univariate.args[1]
+                if degree in poly_dict:
+                    poly_dict[degree] += monomial
+                else:
+                    poly_dict[degree] = monomial
+        elif expr.is_Mul:
+            degree = 0
+            for term in expr.args:
+                term_type = len(term.args)
+                if term_type == 0 and term.is_Symbol:
+                    degree += 1
+                elif term_type == 2:
+                    degree += term.args[1]
+            poly_dict[degree] = expr
+        elif expr.is_Pow:
+            poly_dict[expr.args[1]] = expr
+        else:
+            poly_dict[1] = expr
+    else:
+        poly_dict = {0: expr}
+    return poly_dict
+
+
+def norm(point):
+    """Returns the Euclidean norm of a point from origin.
+    Parameters
+    ==========
+    point: This denotes a point in the dimensional space.
+    Examples
+    ========
+    >>> from sympy.integrals.intpoly import norm
+    >>> from sympy.geometry.point import Point
+    >>> norm(Point(2, 7))
+    sqrt(53)
+    """
+    h = 0
+    half = S(1)/2
+    if isinstance(point, tuple):
+        h = (point[0] ** 2 + point[1] ** 2) ** half
+    elif isinstance(point, Point):
+        h = (point.x ** 2 + point.y ** 2) ** half
+    elif isinstance(point, dict):
+        s = 0
+        for i in point.values():
+            s += i ** 2
+        h = s**half
+    return h
+
+
+def intersection(lineseg_1, lineseg_2):
+    """Returns intersection between lines of which
+    the input line segments are a part of.
+    Note that this function is meant for use in integration_reduction
+    and at that point in the calling function the lines denoted by the
+    segments surely intersect within segment boundaries. Coincident lines
+    are taken to be non-intersecting.
+
+    Parameters
+    ==========
+    lineseg_1, lineseg_2: The input line segments
+
+    Examples
+    ========
+    >>> from sympy.integrals.intpoly import intersection
+    >>> from sympy.geometry.point import Point
+    >>> from sympy.geometry.line import Segment2D
+    >>> l1 = Segment2D(Point(1, 1), Point(3, 5))
+    >>> l2 = Segment2D(Point(2, 0), Point(2, 5))
+    >>> intersection(l1, l2)
+    (2, 3)
+
+    """
+    x1, y1 = lineseg_1.points[0]
+    x2, y2 = lineseg_1.points[1]
+    x3, y3 = lineseg_2.points[0]
+    x4, y4 = lineseg_2.points[1]
+
+    denom = (x1 - x2)*(y3 - y4) - (y1 - y2)*(x3 - x4)
+
+    if denom:
+        t1 = x1*y2 - y1*x2
+        t2 = x3*y4 - x4*y3
+        return((t1*(x3 - x4) - t2*(x1 - x2))/denom,
+               (t1*(y3 - y4) - t2*(y1 - y2))/denom)
+    return ()
+
+
+def is_vertex(ent):
```

Comment:
But how to do it for say `(2, 3)` or `(1, 2, 3)` ? Plus , this function may be expanded in the future.